src/common/Configuration/Config.cpp

```diff
@@ -57,6 +58,76 @@ namespace
 
         return true;
     }
+
+    // Converts ini keys to the environment variable key (upper snake case).
+    // Example of conversions:
+    //   SomeConfig => SOME_CONFIG
+    //   myNestedConfig.opt1 => MY_NESTED_CONFIG_OPT_1
+    //   LogDB.Opt.ClearTime => LOG_DB_OPT_CLEAR_TIME
+    std::string IniKeyToEnvVarKey(std::string const& key)
+    {
+        std::string result = "";
+
+        const char *str = key.c_str();
+        int n = key.length();
+
+        char curr          = 0;
+        bool isEnd         = false;
+        bool nextIsUpper   = false;
+        bool currIsNumeric = false;
+        bool nextIsNumeric = false;
+
+        for (int i = 0; i < n; ++i)
+        {
+            curr = str[i];
+            if (curr == ' ' || curr == '.' || curr == '-')
+            {
+                result += '_';
+                continue;
+            }
+
+            isEnd = i == n - 1;
+            if (!isEnd)
+            {
+                nextIsUpper = isupper(str[i + 1]);
+
+                // handle "aB" to "A_B"
+                if (!isupper(curr) && nextIsUpper)
+                {
+                    result += toupper(curr);
+                    result += '_';
+                    continue;
+                }
+
+                currIsNumeric = isNumeric(curr);
+                nextIsNumeric = isNumeric(str[i + 1]);
+
+                // handle "a1" to "a_1"
+                if (!currIsNumeric && nextIsNumeric)
+                {
+                    result += toupper(curr);
+                    result += '_';
+                    continue;
+                }
+
+                // handle "1a" to "1_a"
+                if (currIsNumeric && !nextIsNumeric)
+                {
+                    result += toupper(curr);
+                    result += '_';
+                    continue;
+                }
+            }
+
+            result += toupper(curr);
+        }
+        return result;
+    }
+
+    char* EnvVarForIniKey(std::string const& key) {
```

Comment:
I would probably do the conversion to `std::string` already here.

Reply:
Would have to return Optional<std::string> then